src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1161,6 +1164,8 @@ public void RemoveFromWorkspace()
                 _eventSubscriptionTracker.Clear();
             }
 
+            _documentFileChangeContext.Dispose();
```

Comment:
Should the dispose be in the lock, or are we guaranteed no race calls to RemoveFromWorkspace?

Reply:
Once we're at this point, we know there's only one thread handling the disposal.